cmd/gateway-gcs.go

```diff
@@ -294,8 +282,48 @@ func (l *gcsGateway) ListBuckets() ([]BucketInfo, error) {
 	return b, nil
 }
 
-// DeleteBucket delete a bucket on GCS
+// DeleteBucket delete a bucket on GCS.
 func (l *gcsGateway) DeleteBucket(bucket string) error {
+	itObject := l.client.Bucket(bucket).Objects(l.ctx, &storage.Query{Delimiter: slashSeparator, Versions: false})
+	// We list the bucket and if we find any objects we return BucketNotEmpty error. If we
+	// find only "minio.sys.temp/" then we remove it before deleting the bucket.
+	gcsMinioPathFound := false
+	nonGCSMinioPathFound := false
+	for {
```

Comment:
@krishnasrinivas My proposal is below
```go
func checkBucketEmpty(objIter ObjectIterator) error {
	for {
		objAttrs, err := objIter.Next()
		if err != nil {
			if err == iterator.Done {
				return nil
			}

			return gcsToObjectError(traceError(err))
		}

		if objAttrs.Prefix != gcsMinioPath {
			return gcsToObjectError(traceError(BucketNotEmpty{}))
		}
	}

	return nil
}
```
it can be used like below
```go
if err = checkBucketEmpty(itObject); err != nil {
    return err
}
```

Here `checkBucketEmpty()` is unit testable.  There is one problem here that if multipart put is initiated in a fresh bucket just after `checkBucketEmpty()`, those data will be removed.  Whereas in current approach, it will error out.

Not sure it makes sense.